src/twisted/mail/imap4.py

```diff
@@ -209,33 +275,76 @@ def clean(self):
         Clean ranges list, combining adjacent ranges
         """
 
-        self.ranges.sort()
+        ranges = sorted((self._swap(low, self._infinity, ifIs=None),
+                         self._swap(high, self._infinity, ifIs=None))
+                         for low, high in self.ranges)
 
-        oldl, oldh = None, None
-        for i,(l, h) in enumerate(self.ranges):
-            if l is None:
+        mergedRanges = [(float('-inf'), float('-inf'))]
```

Comment:
perhaps this should be bound to a global called `_inf`?

(I also vaguely seem to remember that `1e999` is more portable than `inf` as a constant, but none of our buildbots are complaining, so…)